connect/runtime/src/main/java/org/apache/kafka/connect/util/KafkaBasedLog.java

```diff
@@ -281,9 +281,18 @@ private void readToLogEnd() {
             Iterator<Map.Entry<TopicPartition, Long>> it = endOffsets.entrySet().iterator();
             while (it.hasNext()) {
                 Map.Entry<TopicPartition, Long> entry = it.next();
-                if (consumer.position(entry.getKey()) >= entry.getValue())
+                TopicPartition topicPartition = entry.getKey();
+                Long endOffset = entry.getValue();
+                long lastConsumedOffset = consumer.position(topicPartition);
+                if (lastConsumedOffset >= endOffset) {
+                    log.trace("Reached end offset {} for {}", endOffset, topicPartition);
                     it.remove();
-                else {
+                } else {
+                    log.trace(
+                        "Behind end offset {} for {}; last-consumed offset is {}",
+                        endOffset,
+                        topicPartition,
+                        lastConsumedOffset);
```

Comment:
```suggestion
                    log.trace("Behind end offset {} for {}; last-consumed offset is {}",
                            endOffset, topicPartition, lastConsumedOffset);
```
nit: multiline calls don't need to be on their own line in AK and tab is equal to 4 spaces (here we need 2 tabs)

Reply:
Similar to the above, seeing a message that says `read` might be easier to read in context than `consumed`. 
How about: 
`Behind end offset {} for {}; last-read offset is {}`